File: config/src/main/java/org/wcl/mfn/config/ui/common/NavigationBarConfig.java
package org.wcl.mfn.config.ui.common;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.*;

@Configuration
@PropertySource("ui/ui.properties")
public class NavigationBarConfig {
    @Value("${mfn.navbar}")
    private String navigationBarId;

    @Value("${mfn.navbar.home}")
    private String navigationBarHomeId;

    @Value("${mfn.navbar.tools}")
    private String navigationBarToolsId;

    @Value("${mfn.navbar.contract-calculator}")
    private String navigationBarContractCalculatorId;

    public String navigationBarId() {
        return navigationBarId;
    }

    public String navigationBarHomeId() {
        return navigationBarHomeId;
    }

    public String navigationBarToolsId() {
        return navigationBarToolsId;
    }

    public String navigationBarContractCalculatorId() {
        return navigationBarContractCalculatorId;
    }
}

File: config/src/main/kotlin/org/wcl/mfn/config/ui/common/NavigationBarConfig.kt
package org.wcl.mfn.config.ui.common

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.*

@Configuration
@PropertySource("classpath:ui/common/navigation-bar.properties")
open class NavigationBarConfig {
    @Value("\${mfn.navbar}")
    private val navigationBarId: String? = null

    @Value("\${mfn.navbar.home}")
    private val navigationBarHomeId: String? = null

    @Value("\${mfn.navbar.tools}")
    private val navigationBarToolsId: String? = null

    @Value("\${mfn.navbar.contract-calculator}")
    private val navigationBarContractCalculatorId: String? = null

    fun navigationBarId(): String? {
        return navigationBarId
    }

    fun navigationBarHomeId(): String? {
        return navigationBarHomeId
    }

    fun navigationBarToolsId(): String? {
        return navigationBarToolsId
    }

    fun navigationBarContractCalculatorId(): String? {
        return navigationBarContractCalculatorId
    }
}
